libs/constrained-generators/src/Constrained/GenT.hs

```diff
@@ -94,14 +108,15 @@ catGEs (GenError {} : ges) = catGEs ges
 catGEs (FatalError es e : _) =
   runGE $ FatalError es e
 
-fromGE :: ([String] -> a) -> GE a -> a
+fromGE :: (NE.NonEmpty String -> a) -> GE a -> a
 fromGE _ (Result _ a) = a
-fromGE a (GenError es e) = a $ concat es ++ e
+fromGE a (GenError [] e) = a e
+fromGE a (GenError es e) = a $ foldr1 NE.append es `NE.append` e
```

Comment:
I believe you need to use `<>` instead of  `append`, to make 8.10.7 compile
```suggestion
fromGE a (GenError es e) = a $ foldr1 (<>) es <> e
```

Reply:
Yes, and also use pure rather than NE.singleton